### PR TITLE
Restrict routes which consist of a shortest path with duplicate lanelet IDs from being generated

### PR DESCRIPTION
**Pull request by JonSmet**
# PR Details
## Description

This PR includes updates to route_generator_worker in order to check whether a route's shortest path contains multiple occurrences of at least one Lanelet ID. Currently, carma platform does not support this use case, as the occurrence of duplicate Lanelet IDs in a route's shortest path could lead to undefined behavior in other components. For example, if a lanelet occurs twice in a route's shortest path, then it is not clear what downtrack value should be assigned to a given lane centerline point associated with that lanelet (the downtrack associated with the first occurrence of the lanelet in the shortest path, or the second occurrence?). With this PR, a provided route that results in a shortest path consisting of duplicate lanelet IDs will not be generated by route_generator_worker's set_active_route_cb() service callback.

## Related Issue

[Issue #1169: Inlane cruising issues std::bad alloc at tfhrc](https://github.com/usdot-fhwa-stol/carma-platform/issues/1169)

## Motivation and Context

See above.

## How Has This Been Tested?

Integration tested on Black Pacifica and unit tested.

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: route/src/route_generator_worker.cpp [hunk 1]
```diff
@@ -244,6 +253,31 @@ namespace route {
...
+
+        // Iterate through the shortest path to populate shortest_path_lanelet_ids with lanelet IDs
+        for(const auto& ll : route.get().shortestPath())
+        {
+            shortest_path_lanelet_ids.push_back(ll.id());
+        }
+
+        // Verify that there are no duplicate lanelet IDs in the shortest path
+        std::sort(shortest_path_lanelet_ids.begin(), shortest_path_lanelet_ids.end());
+        auto it = std::adjacent_find(shortest_path_lanelet_ids.begin(), shortest_path_lanelet_ids.end());
```
[the remark this answers is not in the document]
Comment (JonSmet) [reply route/src/route_generator_worker.cpp:269]: Thanks for the informative comment and I definitely see the benefit of improving this implementation either here or (more importantly) for a section of the system that is resource-constrained. I'll take note of this for future tasks that are similar.

File: route/src/route_generator_worker.cpp [hunk 2]
```diff
@@ -244,6 +253,31 @@ namespace route {
         return true;
     }
 
+    bool RouteGeneratorWorker::check_for_duplicate_lanelets_in_shortest_path(const lanelet::Optional<lanelet::routing::Route>& route)
```
[the remark this answers is not in the document]
Comment (JonSmet) [reply route/src/route_generator_worker.cpp]: Fixed.

File: route/src/route_generator_worker.cpp [hunk 3]
```diff
@@ -216,6 +216,15 @@ namespace route {
 
             route->setEndPoint(end_point);
 
+            if (check_for_duplicate_lanelets_in_shortest_path(route))
```
[the remark this answers is not in the document]
Comment (JonSmet) [reply route/src/route_generator_worker.cpp]: Fixed.

**Review by msmcconnell — APPROVED** [review]
Looks good. Just waiting for CI

**Pull request merged**